### DISCUSSION TITLE
How to format datetime dd/MM/yy?

### DISCUSSION THREAD

**Discussion by robfolks** [JoshClose/CsvHelper#1897] [Q&A] [answered]
I'm trying to format a datetime in the class map so the output is dd/MM/yy however I can't figure out how to output only the last two digits of the year. 

Map(m => m.InvoiceDate).Index(12).TypeConverterOption.Format("dd/MM/y");
Map(m => m.InvoiceDate).Index(12).TypeConverterOption.Format("dd/MM/yy");
Map(m => m.InvoiceDate).Index(12).TypeConverterOption.Format("dd/MM/yyy");
Map(m => m.InvoiceDate).Index(12).TypeConverterOption.Format("dd/MM/yyyy");
These formats all output a 4 digit code for the year.

Map(m => m.InvoiceDate).Index(11).TypeConverterOption.Format("dd/MM/y hh");
Interestingly if I included the hh then it does output the year as the last two digits but includes the hours in the format which I don't require.

Is anyone able to help with what I'm doing wrong?

**Comment by JoshClose** [comment, marked as answer]
This seems to work fine for me. What type is `InvoiceDate`? What is the culture you're using?

**Reply by robfolks** [reply]
Hi Josh,

Firstly thank you for a great product!

InvoiceDate is a nullable Datetime and culture is en-GB. I am running an older version of CSVHelper 12.1.2.

Rob

**Reply by JoshClose** [reply]
This is works for me. Let me know what is different in yours.

```cs
void Main()
{
    var records = new List<Foo>
    {
        new Foo { Id = 1, Name = "one", Date = DateTime.Now },
    };
    var config = new Configuration(CultureInfo.GetCultureInfo("en-GB"));
    using (var writer = new StringWriter())
    using (var csv = new CsvWriter(writer, config))
    {
        csv.Configuration.RegisterClassMap<FooMap>();
        csv.WriteRecords(records);
        writer.Dump();
    }
}

private class Foo
{
    public int Id { get; set; }
    public string Name { get; set; }
    public DateTime? Date { get; set; }
}

private class FooMap : ClassMap<Foo>
{
    public FooMap()
    {
        Map(m => m.Id).Index(0);
        Map(m => m.Name).Index(1);
        Map(m => m.Date).Index(2).TypeConverterOption.Format("dd/MM/y");
    }
}
```

![image](https://user-images.githubusercontent.com/114971/143467043-a26d926a-74e6-4c82-b162-ae1fa5288fe6.png)

**Reply by robfolks** [reply]
Hi, 

This example works fine for my code too. I must have been naively checking the file in excel which was altering the date field. Sorry for wasting your time this can be closed off.

Rob
